components/ui/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,43 @@
+import { cva, VariantProps } from "class-variance-authority";
+import { Switch } from "@headlessui/react";
+
+const toggleStyles = cva(
+  "relative inline-flex h-[38px] w-[74px] shrink-0 cursor-pointer rounded-full border-2 border-transparent transition-colors duration-200 ease-in-out focus:outline-none focus-visible:ring-2  focus-visible:ring-white focus-visible:ring-opacity-75",
+  {
+    variants: {
+      intent: {
+        lightEnabled: "bg-blue-500",
+        lightDisabled: "bg-blue-200",
+        darkEnabled: "bg-slate-500",
+        darkDisabled: "bg-slate-200",
+      },
+    },
+    defaultVariants: {
+      intent: "lightEnabled",
+    },
+  }
+);
+export interface ToggleProps extends VariantProps<typeof toggleStyles> {}
+
+export const Toggle = ({ intent, ...props }: ToggleProps) => {
+  return (
+    <Switch
+      checked={intent && intent.includes("Enabled") ? true : false}
```

Comment:
with the change above, this logic will be cleaner, i.e. we'll have a `disabled` variant
and can just check if `disabled === true`